src/wallet/rpcdump.cpp

```diff
@@ -775,15 +799,33 @@ UniValue z_exportkey(const UniValue& params, bool fHelp)
     if (!IsValidPaymentAddress(address)) {
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid zaddr");
     }
-    // TODO: Add Sapling support. For now, ensure we can freely convert.
-    assert(boost::get<libzcash::SproutPaymentAddress>(&address) != nullptr);
-    auto addr = boost::get<libzcash::SproutPaymentAddress>(address);
-
-    libzcash::SproutSpendingKey k;
-    if (!pwalletMain->GetSpendingKey(addr, k))
-        throw JSONRPCError(RPC_WALLET_ERROR, "Wallet does not hold private zkey for this zaddr");
-
-    return EncodeSpendingKey(k);
+    // Sapling support
+    bool isValidAddr = boost::get<libzcash::SproutPaymentAddress>(&address) != nullptr || boost::get<libzcash::SaplingPaymentAddress>(&address) != nullptr;
+    assert(isValidAddr);
```

Comment:
There's already an `IsValidPayamentAddress(address)` check above here, so these two lines can be removed.

Reply:
But despite that, there was an assert there before that `address` was not `nullptr`. Why should we remove it now?

Reply:
The assert was there specifically so that we would get a crash if someone tried to use a Sapling address before support was added. Now that `GetSpendingKeyForPaymentAddress` is implemented, we don't need these explicit `nullptr` checks, because if we ever add a new address type, all the `boost::static_visitor`s will immediately start complaining that they aren't covering all cases :slightly_smiling_face:

So please remove these two lines.